src/unix/udp.c

```diff
@@ -135,22 +139,62 @@ static void uv__udp_run_completed(uv_udp_t* handle) {
 }
 
 
+#if defined(__linux__)
+static int uv__udp_recvmsg_errqueue(uv_udp_t* handle,
+                                    struct msghdr* h,
+                                    uv_buf_t* buf,
+                                    const struct sockaddr* peer,
+                                    int flags) {
+  struct cmsghdr* cmsg;
+  struct sock_extended_err* serr;
+
+  if (!(h->msg_flags & MSG_ERRQUEUE))
+    return 0;
+
+  flags |= UV_UDP_LINUX_RECVERR;
+  for (cmsg = CMSG_FIRSTHDR(h); cmsg != NULL; cmsg = CMSG_NXTHDR(h, cmsg)) {
+    if ((cmsg->cmsg_level == SOL_IP   && cmsg->cmsg_type == IP_RECVERR) ||
+        (cmsg->cmsg_level == SOL_IPV6 && cmsg->cmsg_type == IPV6_RECVERR)) {
+      serr = (struct sock_extended_err*) CMSG_DATA(cmsg);
+
+      /* The address of the offending node follows right after serr */
+      struct sockaddr* offender = (struct sockaddr*) (serr + 1);
+
+      handle->recv_cb(handle,
+                      UV__ERR(serr->ee_errno),
+                      buf,
+                      offender != NULL ? offender : peer,
+                      flags);
+      return 1; /* handled */
```

Comment:
Can there be more than one IP_RECVERR or IPV6_RECVERR cmsg?

Reply:
Based on my research, **no**. Each received error generates _just one_ error queue entry.

Refs:
* https://github.com/torvalds/linux/blob/master/net/ipv6/datagram.c#L457C9-L457C24
* https://github.com/torvalds/linux/blob/master/net/ipv6/raw.c#L446
* https://github.com/torvalds/linux/blob/master/net/ipv4/ip_sockglue.c#L525
* https://github.com/torvalds/linux/blob/master/net/ipv4/raw.c#L752